torch/distributed/_shard/api.py

```diff
@@ -265,12 +266,21 @@ def shard_module(
             shard_parameter(
                 mod,
                 param_name,
-                plan.plan[name],
+                spec,
                 src_rank=src_rank,
                 process_group=process_group
             )
+        elif isinstance(spec, Sharder):
```

Comment:
We need some validation here that another item in ShardingPlan doesn't conflict with the submodule tree that Sharder is working with.